judo-runtime-core-jsl-itest/models/RecursiveCompositionModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/automappedto/RecursiveCompositionTest.java

```diff
@@ -0,0 +1,370 @@
+        TransferXAutoMappedTO x8 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x8").build());
+        TransferXAutoMappedTO x5 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x5").build());
+        TransferXAutoMappedTO x3 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x3").build());
+
+        TransferYAutoMappedTO y9 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y9").build());
+        TransferYAutoMappedTO y7 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y7").build());
+        TransferYAutoMappedTO y4 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y4").build());
+        TransferYAutoMappedTO y3 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y3").build());
+        TransferYAutoMappedTO y2 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y2").build());
+        TransferYAutoMappedTO y1 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y1").build());
+
+        TransferYAutoMappedTO y8 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y8").withYx(x13).build());
+        TransferXAutoMappedTO x12 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x12").withY(y9).build());
+        TransferXAutoMappedTO x4 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x4").withY(y4).build());
+
+        TransferXAutoMappedTO x10 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x10").withYs(List.of(y7, y8)).build());
+        TransferXAutoMappedTO x2 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x2").withX(x5).withXs(List.of(x3, x4)).build());
+
+        TransferYAutoMappedTO y6 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y6").withYxs(List.of(x11, x12)).build());
+        TransferYAutoMappedTO y5 = transferYAutoMappedTODao.create(TransferYAutoMappedTO.builder().withName("y5").withYx(x10).build());
+
+        TransferXAutoMappedTO x9 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x9").withYs(List.of(y5, y6)).build());
+        TransferXAutoMappedTO x7 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x7").withXs(List.of(x8, x9)).build());
+        TransferXAutoMappedTO x6 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x6").withX(x7).build());
+
+        TransferXAutoMappedTO x1 = transferXAutoMappedTODao.create(TransferXAutoMappedTO.builder().withName("x1").withX(x2).withXs(List.of(x6)).withY(y1).withYs(List.of(y2, y3)).build());
+
+        assertEquals("x1", x1.getName().orElseThrow());
+        assertEquals("x2", x1.getX().orElseThrow().getName().orElseThrow());
+        assertEquals(1, x1.getXs().stream().count());
```

Comment:
Should be x1..getXs().size()

Reply:
[3832306](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/230/commits/3832306c93a87b2bff4bd38a8af7260dc60cfc5d)